Index: virtualbox/src/VBox/Additions/linux/drm/vbox_fb.c
===================================================================
--- virtualbox.orig/src/VBox/Additions/linux/drm/vbox_fb.c
+++ virtualbox/src/VBox/Additions/linux/drm/vbox_fb.c
@@ -196,9 +196,13 @@
 	.owner = THIS_MODULE,
 	.fb_check_var = drm_fb_helper_check_var,
 	.fb_set_par = drm_fb_helper_set_par,
+#if RTLNX_VER_MIN(6,5,0)
+	__FB_DEFAULT_SYS_OPS_DRAW,
+#else
 	.fb_fillrect = drm_fb_helper_sys_fillrect,
 	.fb_copyarea = drm_fb_helper_sys_copyarea,
 	.fb_imageblit = drm_fb_helper_sys_imageblit,
+#endif
 	.fb_pan_display = drm_fb_helper_pan_display,
 	.fb_blank = drm_fb_helper_blank,
 	.fb_setcmap = drm_fb_helper_setcmap,
