Index: virtualbox/src/VBox/Additions/linux/drm/vbox_drv.c
===================================================================
--- virtualbox.orig/src/VBox/Additions/linux/drm/vbox_drv.c
+++ virtualbox/src/VBox/Additions/linux/drm/vbox_drv.c
@@ -52,7 +52,7 @@
 
 /** Detect whether kernel mode setting is OFF. */
 #if defined(CONFIG_VGA_CONSOLE)
-# if RTLNX_VER_MIN(5,17,0) || RTLNX_RHEL_RANGE(8,7, 8,99) || RTLNX_RHEL_MIN(9,1)
+# if RTLNX_VER_MIN(5,17,0) || RTLNX_RHEL_RANGE(8,7, 8,99) || RTLNX_RHEL_MIN(9,1) || RTLNX_SUSE_MAJ_PREREQ(15,5)
 #  define VBOX_VIDEO_NOMODESET() drm_firmware_drivers_only() && vbox_modeset == -1
 # elif RTLNX_VER_MIN(4,7,0)
 #  define VBOX_VIDEO_NOMODESET() vgacon_text_force() && vbox_modeset == -1
@@ -367,7 +367,7 @@
 #endif
 
 	.fops = &vbox_fops,
-#if RTLNX_VER_MAX(5,15,0) && !RTLNX_RHEL_RANGE(8,7, 8,99) && !RTLNX_RHEL_MAJ_PREREQ(9,1)
+#if RTLNX_VER_MAX(5,15,0) && !RTLNX_RHEL_RANGE(8,7, 8,99) && !RTLNX_RHEL_MAJ_PREREQ(9,1) && !RTLNX_SUSE_MAJ_PREREQ(15,5)
 	.irq_handler = vbox_irq_handler,
 #endif
 	.name = DRIVER_NAME,
Index: virtualbox/src/VBox/Additions/linux/drm/vbox_drv.h
===================================================================
--- virtualbox.orig/src/VBox/Additions/linux/drm/vbox_drv.h
+++ virtualbox/src/VBox/Additions/linux/drm/vbox_drv.h
@@ -157,7 +157,7 @@
 # include <drm/drm_device.h>
 # include <drm/drm_ioctl.h>
 # include <drm/drm_fourcc.h>
-# if RTLNX_VER_MAX(5,15,0) && !RTLNX_RHEL_RANGE(8,7, 8,99) && !RTLNX_RHEL_MAJ_PREREQ(9,1)
+# if RTLNX_VER_MAX(5,15,0) && !RTLNX_RHEL_RANGE(8,7, 8,99) && !RTLNX_RHEL_MAJ_PREREQ(9,1) && !RTLNX_SUSE_MAJ_PREREQ(15,5)
 #  include <drm/drm_irq.h>
 # endif
 # include <drm/drm_vblank.h>
@@ -189,7 +189,7 @@
 # include <drm/ttm/ttm_resource.h>
 #endif
 
-#if RTLNX_VER_MIN(6,0,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,2)
+#if RTLNX_VER_MIN(6,0,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,2) || RTLNX_SUSE_MAJ_PREREQ(15,5)
 # include <drm/drm_framebuffer.h>
 #endif
 
@@ -536,7 +536,7 @@
 int vbox_irq_init(struct vbox_private *vbox);
 void vbox_irq_fini(struct vbox_private *vbox);
 void vbox_report_hotplug(struct vbox_private *vbox);
-#if RTLNX_VER_MAX(5,15,0) && !RTLNX_RHEL_MAJ_PREREQ(9,1)
+#if RTLNX_VER_MAX(5,15,0) && !RTLNX_RHEL_MAJ_PREREQ(9,1) && !RTLNX_SUSE_MAJ_PREREQ(15,5)
 irqreturn_t vbox_irq_handler(int irq, void *arg);
 #endif
 
Index: virtualbox/src/VBox/Additions/linux/drm/vbox_fb.c
===================================================================
--- virtualbox.orig/src/VBox/Additions/linux/drm/vbox_fb.c
+++ virtualbox/src/VBox/Additions/linux/drm/vbox_fb.c
@@ -378,7 +378,7 @@
 	info->screen_size = size;
 
 #ifdef CONFIG_FB_DEFERRED_IO
-# if RTLNX_VER_MIN(5,19,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,3)
+# if RTLNX_VER_MIN(5,19,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,3) || RTLNX_SUSE_MAJ_PREREQ(15,5)
 	info->fix.smem_len = info->screen_size;
 # endif
 	info->fbdefio = &vbox_defio;
Index: virtualbox/src/VBox/Additions/linux/drm/vbox_irq.c
===================================================================
--- virtualbox.orig/src/VBox/Additions/linux/drm/vbox_irq.c
+++ virtualbox/src/VBox/Additions/linux/drm/vbox_irq.c
@@ -205,7 +205,7 @@
 {
 	INIT_WORK(&vbox->hotplug_work, vbox_hotplug_worker);
 	vbox_update_mode_hints(vbox);
-#if RTLNX_VER_MIN(5,15,0) || RTLNX_RHEL_RANGE(8,7, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,1)
+#if RTLNX_VER_MIN(5,15,0) || RTLNX_RHEL_RANGE(8,7, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,1) || RTLNX_SUSE_MAJ_PREREQ(15,5)
 	return request_irq(VBOX_DRM_TO_PCI_DEV(vbox->dev)->irq, vbox_irq_handler, IRQF_SHARED, vbox->dev->driver->name, vbox->dev);
 #elif RTLNX_VER_MIN(3,16,0) || RTLNX_RHEL_MAJ_PREREQ(7,1)
 	return drm_irq_install(vbox->dev, VBOX_DRM_TO_PCI_DEV(vbox->dev)->irq);
@@ -216,7 +216,7 @@
 
 void vbox_irq_fini(struct vbox_private *vbox)
 {
-#if RTLNX_VER_MIN(5,15,0) || RTLNX_RHEL_RANGE(8,7, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,1)
+#if RTLNX_VER_MIN(5,15,0) || RTLNX_RHEL_RANGE(8,7, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,1) || RTLNX_SUSE_MAJ_PREREQ(15,5)
 	free_irq(VBOX_DRM_TO_PCI_DEV(vbox->dev)->irq, vbox->dev);
 #else
 	drm_irq_uninstall(vbox->dev);
Index: virtualbox/src/VBox/Additions/linux/drm/vbox_mode.c
===================================================================
--- virtualbox.orig/src/VBox/Additions/linux/drm/vbox_mode.c
+++ virtualbox/src/VBox/Additions/linux/drm/vbox_mode.c
@@ -50,7 +50,7 @@
 # include <drm/drm_probe_helper.h>
 #endif
 
-#if RTLNX_VER_MIN(6,0,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,2)
+#if RTLNX_VER_MIN(6,0,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,2) || RTLNX_SUSE_MAJ_PREREQ(15,5)
 # include <drm/drm_edid.h>
 #endif
 
Index: virtualbox/src/VBox/Additions/linux/drm/vbox_ttm.c
===================================================================
--- virtualbox.orig/src/VBox/Additions/linux/drm/vbox_ttm.c
+++ virtualbox/src/VBox/Additions/linux/drm/vbox_ttm.c
@@ -321,7 +321,7 @@
 #if RTLNX_VER_MAX(5,10,0) && !RTLNX_RHEL_RANGE(8,5, 8,99)
 	tt->func = &vbox_tt_backend_func;
 #endif
-#if RTLNX_VER_MIN(5,19,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,2)
+#if RTLNX_VER_MIN(5,19,0) || RTLNX_RHEL_RANGE(8,8, 8,99) || RTLNX_RHEL_MAJ_PREREQ(9,2) || RTLNX_SUSE_MAJ_PREREQ(15,5)
 	if (ttm_tt_init(tt, bo, page_flags, ttm_write_combined, 0)) {
 #elif RTLNX_VER_MIN(5,11,0) || RTLNX_RHEL_RANGE(8,5, 8,99)
 	if (ttm_tt_init(tt, bo, page_flags, ttm_write_combined)) {
@@ -628,7 +628,7 @@
 	drm_vma_node_reset(&vboxbo->bo.base.vma_node);
 #endif
 
-#if RTLNX_VER_MIN(6,1,0) || RTLNX_RHEL_MAJ_PREREQ(9,3)
+#if RTLNX_VER_MIN(6,1,0) || RTLNX_RHEL_MAJ_PREREQ(9,3) || RTLNX_SUSE_MAJ_PREREQ(15,5)
 	ret = ttm_bo_init_validate(&vbox->ttm.bdev, &vboxbo->bo,
 #else
 	ret = ttm_bo_init(&vbox->ttm.bdev, &vboxbo->bo, size,
