Index: include/iprt/string.h
===================================================================
--- a/include/iprt/string.h	(revision 102991)
+++ b/include/iprt/string.h	(revision 102992)
@@ -255,16 +255,16 @@
  *
  * @param   a_pDst          Pointer to the destination string buffer.
  * @param   a_pSrc          Pointer to the source NULL-terminated string buffer.
- * @param   a_cbToCopy      Size of destination buffer..
+ * @param   a_cbDst         Size of destination buffer.
  */
 #if defined(RT_OS_LINUX) && defined(__KERNEL__)
 # if (RTLNX_VER_MIN(3,16,60))
-#  define RT_STRSCPY(a_pDst, a_pSrc, a_cbToCopy)    strscpy((a_pDst), (a_pSrc), (a_cbToCopy))
+#  define RT_STRSCPY(a_pDst, a_pSrc, a_cbDst)   (void)strscpy((a_pDst), (a_pSrc), (a_cbDst))
 # else /* < 3.16.60 */
-#  define RT_STRSCPY(a_pDst, a_pSrc, a_cbToCopy)    strlcpy((a_pDst), (a_pSrc), (a_cbToCopy))
+#  define RT_STRSCPY(a_pDst, a_pSrc, a_cbDst)   strlcpy((a_pDst), (a_pSrc), (a_cbDst))
 # endif
 #else  /* !RT_OS_LINUX && !__KERNEL__ */
-# define RT_STRSCPY(a_pDst, a_pSrc, a_cbToCopy)    strscpy((a_pDst), (a_pSrc), (a_cbToCopy))
+# define RT_STRSCPY(a_pDst, a_pSrc, a_cbDst)    (void)strscpy((a_pDst), (a_pSrc), (a_cbDst))
 #endif /* !RT_OS_LINUX && !__KERNEL__ */
 
 
